Tests/SonarScanner.MSBuild.PreProcessor.Test/EngineResolution/EngineResolverTests.cs

```diff
@@ -193,7 +194,7 @@ public async Task ResolveEngine_DownloadSucceedsInRetry()
     }
 
     [TestMethod]
-    public async Task ResolveEngine_MetadataDownload_DoesNotRetry()
+    public async Task ResolveEngine_MetadataDownloadSucceedsInRetry()
```

Comment:
Add a UT where the second download also fails, so we make sure we do not end up in an endless loop and we can assert `TelemetryValues.ScannerEngineDownload.Failed`